rmkdir

critBus opened the pull request:
Básicamente agregue un método nuevo a la interfaz com.nerox.client.callbacks.ITfprotocolCallback
 
```
default void rmkdirCallback(StatusInfo status){
        throw new RuntimeException("Callback is not implemented: exception");
    }
```

Y su correspondiente método a  com.nerox.client.Tfprotocol

```
public void rmkdirCommand(String path) {
        this.getProtoHandler().rmkdirCallback(
                this.easyreum.getBuilder().build("RMKDIR",path).translate()
                        .getBuilder().buildStatusInfo());
    }
```

critBus committed:
rmkdir

godjangollc commented on src/main/java/com/nerox/client/callbacks/ITfprotocolCallback.java:
```diff
@@ -18,6 +18,11 @@ default void mkdirCallback(StatusInfo status){
         throw new RuntimeException("Callback is not implemented: exception");
     }
 
+    //rene
```
This comment is not related to the code, please remove it

godjangollc replied:
As you created a new function which by the way adds a new feature, please include as well a new unit testing to the current existing unit testing... 
Please read the following [link](https://www.conventionalcommits.org/en/v1.0.0/) to improve the commits content. Also, remember this is a public repo not only for Hispano speakers but for the whole world. Please use English for comments, code signatures, and everywhere... thanks. I'm hoping for your next commit for review it.

godjangollc replied:
@critBus @lmdelbahia review the comments above

critBus committed:
test: Test for RMKDIR command

Add test com.nerox.client.RmkdirCommandTest which attempts to create a folder on the server to test the functionality of the new RMKDIR command added

godjangollc commented on src/test/java/com/nerox/client/RmkdirCommandTest.java:
Why did you create a whole new file for the test?? please use the already created file with unit tests and add the tests to that file, use [AppTest.java](https://github.com/GoDjango-Development/TFProtocolClientJava/blob/main/src/test/java/com/nerox/client/AppTest.java), please notice that we use a different java class for each module, not each command... I think that's the last change we need to approve the commit. In the logic of the test please don't set in the same function that amount of steps because you can have errors for connecting, for invalid keys, for server errors, and a lot more errors not related to the command, as a result, we use a different approach, we do the connection in one function, and then we use the command in another one, for this particular command do client side validations and then assert to a certain condition to be true, then do server-side validation(i.e. stat command, etc.) .... Try to cover as most as possible in the units testings for each command... It will always be welcome more than one unit of testing per command... best regards

godjangollc commented on src/main/java/com/nerox/client/callbacks/ITfprotocolCallback.java:
This file is OK, please don't add any changes to it

godjangollc commented on .gitignore:
When publishing to a public repo, please refrain from uploading changes not related to the fix or feat that is being solved. This causes the reviewer (in this case me) to review more codes... I would appreciate if you delete this from the PR... thanks beforehand

godjangollc reviewed (changes requested):
The requested changes are for: 
Keep a clean structure in the code
Delete changes not related to the PR (i.e .gitignore changes)

critBus committed:
test: Modification of the test for the MKDIR command

The test is included within the com.nerox.client.AppTest class, represented by the testsRmkdir method
The previous test that was represented by the com.nerox.client.RmkdirCommandTest class is eliminated
The .gitignore file is restored

godjangollc reviewed (approved):
Done and thanks you very much
